presto-main/src/main/java/com/facebook/presto/execution/SqlTaskExecution.java

```diff
@@ -258,6 +259,14 @@ private SqlTaskExecution(
         }
     }
 
+    private void stopAllDriverExecution()
+    {
+        taskExecutor.removeTask(taskHandle);
+        for (DriverFactory factory : allRawDriverFactories) {
+            factory.noMoreDrivers();
+        }
```

Comment:
I'm trying to reason about the implication of this out-of-bound closure of DriverFactory. That's when I found this issue.

I believe a commit from 2013 (895c62a9a8283eb7944cd2544ecb38878553fd94) introduced a bug that makes DriverFactory closure not guaranteed.

In `DriverSplitRunner.close`, it should decrement the `pendingCreation` counter in `DriverSplitRunnerFactory` if `driver == null`.